server/src/main/java/io/druid/client/cache/LocalCacheProvider.java

```diff
@@ -33,13 +40,60 @@
   @Min(0)
   private int initialSize = 500000;
 
-  @JsonProperty
-  @Min(0)
-  private int logEvictionCount = 0;
+  @JacksonInject
+  private DruidProcessingConfig config = null;
 
   @Override
   public Cache get()
   {
-    return new MapCache(new ByteCountingLRUMap(initialSize, logEvictionCount, sizeInBytes));
+    if(sizeInBytes > 0) {
+      return MapCache.create(sizeInBytes, initialSize, config.getNumThreads());
```

Comment:
as per https://github.com/druid-io/druid/pull/1849#issuecomment-151696915 only in java 7 and earlier.